File: GeneratedEntityReaderClasses/Components/LifetimeComponent.cs
namespace GeneratedNoitaClasses;
public class LifetimeComponent : NoitaComponentBase,  iNoitaType<LifetimeComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaInt lifetime { get; set; } = new();
	public ValueRange randomize_lifetime { get; set; } = new();
	public NoitaBool fade_sprites { get; set; } = new();
	public NoitaBool kill_parent { get; set; } = new();
	public NoitaBool kill_all_parents { get; set; } = new();
	public NoitaBool serialize_duration { get; set; } = new();
	public NoitaInt kill_frame_serialized { get; set; } = new();
	public NoitaInt creation_frame_serialized { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		lifetime.Read(s);
		randomize_lifetime.Read(s);
		fade_sprites.Read(s);
		kill_parent.Read(s);
		kill_all_parents.Read(s);
		serialize_duration.Read(s);
		kill_frame_serialized.Read(s);
		creation_frame_serialized.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		lifetime.Write(s);
		randomize_lifetime.Write(s);
		fade_sprites.Write(s);
		kill_parent.Write(s);
		kill_all_parents.Write(s);
		serialize_duration.Write(s);
		kill_frame_serialized.Write(s);
		creation_frame_serialized.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/LifetimeComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt
from GeneratedClasses.ValueRange import ValueRange

from NoitaStream import NoitaStream
@dataclass
class LifetimeComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    lifetime:NoitaInt = field(default_factory=NoitaInt)
    randomize_lifetime:ValueRange = field(default_factory=ValueRange)
    fade_sprites:NoitaBool = field(default_factory=NoitaBool)
    kill_parent:NoitaBool = field(default_factory=NoitaBool)
    kill_all_parents:NoitaBool = field(default_factory=NoitaBool)
    serialize_duration:NoitaBool = field(default_factory=NoitaBool)
    kill_frame_serialized:NoitaInt = field(default_factory=NoitaInt)
    creation_frame_serialized:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.lifetime.Read(s)
        self.randomize_lifetime.Read(s)
        self.fade_sprites.Read(s)
        self.kill_parent.Read(s)
        self.kill_all_parents.Read(s)
        self.serialize_duration.Read(s)
        self.kill_frame_serialized.Read(s)
        self.creation_frame_serialized.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.lifetime.Write(s)
        self.randomize_lifetime.Write(s)
        self.fade_sprites.Write(s)
        self.kill_parent.Write(s)
        self.kill_all_parents.Write(s)
        self.serialize_duration.Write(s)
        self.kill_frame_serialized.Write(s)
        self.creation_frame_serialized.Write(s)

